packages/token-providers/src/fromEnvBearerToken.ts

```diff
@@ -0,0 +1,34 @@
+import { getBearerTokenEnvKey } from "@aws-sdk/core";
+import type { CredentialProviderOptions, TokenIdentityProvider } from "@aws-sdk/types";
+import { TokenProviderError } from "@smithy/property-provider";
+
+export interface FromEnvBearerTokenInit extends CredentialProviderOptions {
+  signingName?: string;
+}
+
+/**
+ * Creates a TokenIdentityProvider that retrieves bearer token from environment variable
+ *
+ * @param options - Configuration options for the token provider
+ * @param options.logger - Optional logger for debug messages
+ * @param options.signingName - Service signing name used to determine environment variable key
+ * @returns TokenIdentityProvider that provides bearer token from environment variable
+ *
+ * @public
+ */
+export const fromEnvBearerToken =
+  ({ logger, signingName }: FromEnvBearerTokenInit = {}): TokenIdentityProvider =>
+  async () => {
+    logger?.debug("@aws-sdk/token-providers - fromEnvBearerToken");
```

Comment:
```suggestion
    logger?.debug?.("@aws-sdk/token-providers - fromEnvBearerToken");
```

Reply:
This came up before, where JS users would not provide a full logger implementation and run into errors.